tests/integration/tests/conftest.py

```diff
@@ -25,6 +25,28 @@ def _harness_clean(h: harness.Harness):
         h.cleanup()
 
 
+def _generate_inspection_report(h: harness.Harness, instance_id: str):
+    LOG.debug("Generating inspection report for %s", instance_id)
+
+    inspection_path = Path(config.INSPECTION_REPORTS_DIR)
+    result = h.exec(
+        instance_id,
+        ["/snap/k8s/current/k8s/scripts/inspect.sh", "/inspection-report.tar.gz"],
+        capture_output=True,
+        text=True,
+    )
+
+    (inspection_path / instance_id).mkdir(parents=True, exist_ok=True)
+    with open(inspection_path / instance_id / "inspection_report_logs.txt", "w") as f:
```

Comment:
you can also use (path).write_text()

Reply:
that's actually pretty cool - thanks TIL!